com.github.Anuken.Mindustry.yml

```
@@ -18,43 +18,48 @@ finish-args:
 
 build-options:
   env:
-    JAVA_HOME: /usr/lib/sdk/openjdk11/
+    JAVA_HOME: /usr/lib/sdk/openjdk17/
 modules:
   - name: openjdk
     buildsystem: simple
     build-commands:
-      - /usr/lib/sdk/openjdk11/install.sh
+      - /usr/lib/sdk/openjdk17/install.sh
 
-
-  # We don't have a gradle generator just yet, so we don't build from source
-  - name: mindustry-binary
+  - name: mindustry
     buildsystem: simple
-    build-commands:
-      - install -D Mindustry.jar /app/bin/Mindustry.jar
-      - install -D mindustry.sh /app/bin/mindustry.sh
-      - install -D com.github.Anuken.Mindustry.desktop /app/share/applications/com.github.Anuken.Mindustry.desktop
-      - install -D com.github.Anuken.Mindustry.metainfo.xml /app/share/metainfo/com.github.Anuken.Mindustry.metainfo.xml
-      - install -D com.github.Anuken.Mindustry.png /app/share/icons/hicolor/512x512/apps/com.github.Anuken.Mindustry.png
-
     sources:
+      # Main repo
+      - type: git
+        url: 'https://github.com/Anuken/Mindustry'
+        tag: 'v135'
+        commit: c24c5c6196cd89bdc17a079a2a8acd372cc260d9
```

This should enable x-checker-data.

```suggestion
        commit: c24c5c6196cd89bdc17a079a2a8acd372cc260d9
        x-checker-data:
          type: git
          tag-pattern: "^v([\\d.]+)$"
```

Can confirm it works:

```
TheEvilSkeleton@TheMainDesktop ~/P/com.github.Anuken.Mindustry (beta)> flatpak-external-data-checker com.github.Anuken.Mindustry.yml
INFO    src.manifest: Checking 2 external data items
INFO    src.manifest: Skipped check [1/2] file com.github.Anuken.Mindustry.png (from com.github.Anuken.Mindustry.yml)
INFO    src.manifest: Started check [2/2] git Mindustry (from com.github.Anuken.Mindustry.yml)
INFO    src.lib.externaldata: Source Mindustry: got new version 135
INFO    src.manifest: Finished check [2/2] git Mindustry (from com.github.Anuken.Mindustry.yml)
CHANGE SOON: Mindustry
 Has a new version:
  URL:       https://github.com/Anuken/Mindustry
  Commit:    c24c5c6196cd89bdc17a079a2a8acd372cc260d9
  Tag:       v135
  Branch:    None
  Version:   135
  Timestamp: None
```

---

Oops :sweat_smile: https://github.com/flathub/com.github.Anuken.Mindustry/pull/39/commits/1a6417f3595bd16697984617e72d7167fb8dc038

---

Oh, yeah makes sense.

---

Should I remove the `tag:` too?

---

I prefer if we keep it for semantics.